### PR TITLE
feat(grok): add kubernetes groks

### PR DESCRIPTION
**Pull request by dominic-mcallister-logdna**
summary: add a few groks to suppose parsing of k8s entities. Add specific grok to parse the k8s filename from the mezmo k8s agent

ref: LOG-18911

### CODE REVIEW

File: src/stdlib/parse_grok.rs [hunk 1]
```diff
@@ -17,7 +17,12 @@ static MEZMO_PATTERNS: &[(&str, &str)] = &[
     ("OPTIONAL_SENTENCE", r#"[\p{L},":;\s\-]*"#),
     ("PSQL_LOG", "%{TIMESTAMP_ISO8601:timestamp} %{DATA} %{SQUARE_BRACKET}%{INT:pid}%{SQUARE_BRACKET} %{EMAILLOCALPART:username}%{NOTSPACE}%{WORD:database} %{GREEDYDATA:message}"),
     ("GOLANG_LOG", "%{DATESTAMP:timestamp} %{LOGLEVEL:level} %{GREEDYDATA:message}"),
-    ("ELASTICSEARCH_LOG", "%{SQUARE_BRACKET}%{TIMESTAMP_ISO8601:timestamp}%{SQUARE_BRACKET}%{SQUARE_BRACKET}%{LOGLEVEL:level}%{SPACE}%{SQUARE_BRACKET}%{SQUARE_BRACKET}%{DATA:source}%{SQUARE_BRACKET} %{GREEDYDATA:message}")
+    ("ELASTICSEARCH_LOG", "%{SQUARE_BRACKET}%{TIMESTAMP_ISO8601:timestamp}%{SQUARE_BRACKET}%{SQUARE_BRACKET}%{LOGLEVEL:level}%{SPACE}%{SQUARE_BRACKET}%{SQUARE_BRACKET}%{DATA:source}%{SQUARE_BRACKET} %{GREEDYDATA:message}"),
+    ("K8S_POD_NAME", r"[a-z0-9A-Z\-.]+"),
```
Comment (biblicalph) [inline src/stdlib/parse_grok.rs:21]: Given that pod, namespace and container patterns are identical, can we define a single pattern and re-use to extract the k8s resource name?

**Review by biblicalph — COMMENTED** [review]
One question

File: src/stdlib/parse_grok.rs (hunk 1 above)
Comment (dominic-mcallister-logdna) [reply src/stdlib/parse_grok.rs:21]: My thought process was each type is specific regardless of them matching via the same regex. That said, perhaps I could define something like a `%{NUMBERS_LETTERS_DASH}`to share amongst the three? I'm not sure how useful it'll be outside this context however.

Comment (biblicalph) [reply src/stdlib/parse_grok.rs:21]: Each pattern has to be compiled at last once. 
We can definitely keep the existing names `K8S_POD_NAME`, `K8S_NAMESPACE`, `K8S_CONTAINER` but they should reference the same pattern

**Review by biblicalph — APPROVED** [review]

**Pull request merged**